core/domain/user_services.py

```diff
@@ -352,16 +352,18 @@ def get_users_settings(
     return result
 
 
-def generate_initial_profile_picture(user_id: str) -> None:
+def generate_initial_profile_picture(user_email: str) -> str:
     """Generates a profile picture for a new user and
     updates the user's settings in the datastore.
 
     Args:
-        user_id: str. The unique ID of the user.
+        user_email: str. The email of the user.
+
+    Returns:
+        str. The profile picture data url.
     """
-    user_email = get_email_from_user_id(user_id)
     user_gravatar = fetch_gravatar(user_email)
-    update_profile_picture_data_url(user_id, user_gravatar)
+    return user_gravatar
```

Comment:
Let's just get rid of this function altogether and have the caller call fetch_gravatar() directly.

Reply:
Done